src/shared/ui/switch/switch-trigger.tsx

```diff
@@ -0,0 +1,37 @@
+import { cva } from 'class-variance-authority'
+
+import { cn } from '@/shared/utils'
+
+import { useSwitchContext } from './switch-context'
+
+const triggerVariants = cva('w-5 h-5 rounded-full bg-gray-0 block transition-transform duration-200', {
+  variants: {
+    variant: {
+      true: 'translate-x-full',
+      false: 'translate-x-0',
+    },
+  },
+  defaultVariants: {
+    variant: false,
+  },
+})
+
+export default function SwitchTrigger() {
+  const { isChecked, setIsChecked } = useSwitchContext()
+
+  return (
+    <label className="rounded-full bg-gray-100 p-1 w-12 h-8 flex items-center cursor-pointer" htmlFor="switch-trigger">
+      <input
+        type="checkbox"
+        className="sr-only"
+        id="switch-trigger"
+        role="switch"
+        aria-checked={isChecked}
+        aria-label="Switch"
+        onChange={() => setIsChecked(!isChecked)}
```

Comment:
p2
set 함수로 토글 방식을 다룰 때는 이전 상태 값을 기준으로 변경 시켜주는게 좋을 것 같아요!
```suggestion
        onChange={() => setIsChecked(prevIsChecked => !prevIsChecked)} // 변수명은 자유롭게
```

Reply:
> p2 set 함수로 토글 방식을 다룰 때는 이전 상태 값을 기준으로 변경 시켜주는게 좋을 것 같아요!

~~useControllableState에서 어느정도 처리가되어서 안전할거라고 기대했는데, 아니군요! 함수형 업데이트로 수정하겠습니다~~

이건 useControllableState를 봐보시면, 제어 컴포넌트로 사용하게 될 경우 말씀하신것처럼 사용할 수 있어요. 그게 아니라면, useUnControlledState에서 useEffect와 useRef로 항상 최신의 값을 반영하고 있어서, 함수형 업데이트를 쓸 필요는 없어보여요